Modules/rat_command.py

```diff
@@ -205,12 +170,20 @@ def real_decorator(func):
         return real_decorator
 
     @classmethod
-    def get_command(cls, name: str):
-        # remove the prefix.
-        name = name.strip(cls.prefix)
-        # see if its a command
-        if name in cls._registered_commands:
-            cls.log.debug("command found!")
-            return cls._registered_commands[name]
-        else:
-            return None
+    def rule(cls, regex: str):
```

Comment:
I do not think all commands hooked via a rule should receive the event if no conventional commands were found, though perhaps a wildcard regex would suffice?

Reply:
I'm afraid I don't quite follow you?

Reply:
Only one matching rule is executed if no conventional command was found. Since python dicts from 3.6 onward are more or less ordered, this will be the first registered rule.

Reply:
ahh, so its First *matching* rule. ok.

That means i misunderstood. carry on